spec/features/products_form_spec.rb

```diff
@@ -53,13 +59,29 @@
     ProductDashboard::ATTRIBUTE_TYPES[:release_year] = old_release_year
   end
 
-  it "edits product and meta tag data correctly" do
-    product = create(:product)
+  it "edits product and meta tag data correctly", js: true do
+    product = create(:product, banner: <<~HTML)
+      <div>A banner with a <a href="https://example.com">link</a>.</div>
+    HTML
 
     visit edit_admin_product_path(product)
 
+    within(:rich_text_area, "Banner") do
+      within("div", text: "A banner with a link") do
+        expect(page).to have_link("link", href: "https://example.com")
+      end
+    end
+
+    fill_in_rich_text_area "Banner", with: <<~HTML
+      <div>A changed banner without a link.</div>
+    HTML
     click_on "Update Product"
 
+    within(".trix-content") do
+      within("div", text: "A changed banner without a link.") do
+        expect(page).to have_no_link(href: "https://example.com")
+      end
+    end
```

Comment:
This can be collapsed as well:

```suggestion
    within(".trix-content div", text: "A changed banner without a link.") do
      expect(page).to have_no_link(href: "https://example.com")
    end
```